owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.kt

```diff
@@ -0,0 +1,451 @@
+            text: String,
+            mimeType: String,
+            tabLayout: TabLayout,
+            viewPager: ViewPager2
+        ) {
+            val adapter = PreviewFormatTextFragmentStateAdapter(this@PreviewTextFragment, text, mimeType)
+            viewPager.adapter = adapter
+
+            TabLayoutMediator(tabLayout, viewPager) { tab: TabLayout.Tab, position: Int ->
+                if (position == 0) {
+                    tab.text = adapter.formatTypes[mimeType]
+                } else {
+                    tab.text = adapter.formatTypes[PreviewFormatTextFragmentStateAdapter.TYPE_PLAIN]
+                }
+            }.attach()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.file_actions_menu, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+
+        if (mContainerActivity.storageManager != null) {
+            val safeFile = file
+            val accountName = mContainerActivity.storageManager.account.name
+            previewTextViewModel.filterMenuOptions(safeFile, accountName)
+
+            collectLatestLifecycleFlow(previewTextViewModel.menuOptions) { menuOptions ->
+                val hasWritePermission = safeFile.hasWritePermission
+                menu.filterMenuOptions(menuOptions, hasWritePermission)
+            }
+        }
+
+        menu.findItem(R.id.action_search)?.apply {
+            isVisible = false
+            isEnabled = false
+        }
+
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_share_file -> {
+                mContainerActivity.fileOperationsHelper.showShareFile(file)
+                return true
+            }
+
+            R.id.action_open_file_with -> {
+                openFile()
+                return true
+            }
+
+            R.id.action_see_details -> {
+                seeDetails()
+                return true
+            }
+
+            R.id.action_send_file -> {
+                requireActivity().sendDownloadedFilesByShareSheet(listOf(file))
+                return true
+            }
+
+            R.id.action_set_available_offline -> {
+                val fileToSetAsAvailableOffline = ArrayList<OCFile>()
+                fileToSetAsAvailableOffline.add(file)
+                fileOperationsViewModel.performOperation(SetFilesAsAvailableOffline(fileToSetAsAvailableOffline))
+                return true
+            }
+
+            R.id.action_unset_available_offline -> {
+                val fileToUnsetAsAvailableOffline = ArrayList<OCFile>()
+                fileToUnsetAsAvailableOffline.add(file)
+                fileOperationsViewModel.performOperation(FileOperation.UnsetFilesAsAvailableOffline(fileToUnsetAsAvailableOffline))
+                return true
+            }
```

Comment:
You forgot to add here the logic for `action_remove_file` and `action_sync_file` actions!

Reply:
I added those two cases at the same place that were in legacy ✅